includes/compatibility.php

```diff
@@ -229,37 +231,71 @@ public static function filter_library_post_type( $post_types ) {
 	}
 
 	/**
-	 * Polylang compatibility.
+	 * Set editor language.
 	 *
-	 * Fix Polylang compatibility with Elementor.
-	 *
-	 * @since 2.0.0
+	 * @since 3.7.0
 	 * @access private
 	 * @static
 	 */
-	private static function polylang_compatibility() {
-		// Fix language if the `get_user_locale` is difference from the `get_locale
+	private static function set_editor_language() {
 		if ( isset( $_REQUEST['action'] ) && 0 === strpos( $_REQUEST['action'], 'elementor' ) ) {
 			add_action( 'set_current_user', function() {
 				global $current_user;
-				$current_user->locale = get_locale();
-			} );
 
-			// Fix for Polylang
-			define( 'PLL_AJAX_ON_FRONT', true );
+				$elementor_preferences = get_user_meta( $current_user->ID, 'elementor_preferences' );
+				$user_interface = isset( $elementor_preferences[0]['user_interface'] ) ? $elementor_preferences[0]['user_interface'] : null;
 
-			add_action( 'pll_pre_init', function( $polylang ) {
-				if ( isset( $_REQUEST['post'] ) ) {
-					$post_language = $polylang->model->post->get_language( $_REQUEST['post'], 'locale' );
-					if ( ! empty( $post_language ) ) {
-						$_REQUEST['lang'] = $post_language->locale;
+				if ( 'user_language' === $user_interface ) {
+					self::wpml_set_editor_language( $user_interface );
+				} else {
+					if ( ! self::wpml_set_editor_language( $user_interface ) ) {
+						$current_user->locale = get_locale();
 					}
 				}
 			} );
+
```

Comment:
I don't think we need this code anymore, but I wasn't sure if it does something that I am unaware of.

Reply:
Its purpose was to fix a bug for Polylang, I checked again and Polylang is working good with the new code.